src/api/app/models/local_build_result.rb

```diff
@@ -0,0 +1,31 @@
+class LocalBuildResult
+  include ActiveModel::Model
+  attr_accessor :project, :package, :repository, :architecture, :code
+
+  def self.find_by(opts)
```

Comment:
I would use another name, to avoid confusion with the `find_by` rails method. I would also add all the parameters explicitly. Something like:

``` Ruby
def self.find_by_opts(project, package, repository, architecture)
```